### PR TITLE
Build and install aws-sdk-cpp in wheel images

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
Contributes to https://github.com/rapidsai/kvikio/pull/426

### CODE REVIEW

File: ci-wheel.Dockerfile [hunk 1]
```diff
@@ -119,6 +119,25 @@ mv gh_*/bin/gh /usr/local/bin
...
+git clone --recurse-submodules -b ${AWS_SDK_CPP_VER} https://github.com/aws/aws-sdk-cpp.git
+cd aws-sdk-cpp
+cmake \
+  -S . \
+  -B build \
+  -DCMAKE_INSTALL_PREFIX=/usr/local \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DBUILD_ONLY=s3 \
+  -DBUILD_SHARED_LIBS=OFF \
+  -DENABLE_TESTING=OFF
```
Comment (jameslamb) [inline ci-wheel.Dockerfile:135]:
```suggestion
  -DENABLE_TESTING=OFF \
  -DENABLE_UNITY_BUILD=ON
```

What do you think about enabling this option?

From https://github.com/aws/aws-sdk-cpp/blob/af521afd75449b7617cb27e8fff6adf67e35d613/CMakeLists.txt#L31

> _"If enabled, the SDK will be built using a single unified .cpp file for each service library.  Reduces the size of static library binaries on Windows and Linux"_

That seems useful for our purposes.

**Review by jameslamb — APPROVED** [review]
Given the choice to build this from source in these images, the implementation looks good to me. Thanks for also adding it to the `renovate` config!

I left one small suggestion for your consideration. Do what you want with it, and I don't need to review again.

File: ci-wheel.Dockerfile (hunk 1 above)
Comment (KyleFromNVIDIA) [reply ci-wheel.Dockerfile]: Good idea. I've turned on unity build - we'll see how well it works.

**Review by vyasr — APPROVED** [review]
Looks fine, although I didn't review the full set of available build options for aws-sdk-cpp.

File: ci-wheel.Dockerfile [hunk 2]
```diff
@@ -119,6 +119,26 @@ mv gh_*/bin/gh /usr/local/bin
...
+  -B build \
+  -DCMAKE_INSTALL_PREFIX=/usr/local \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DBUILD_ONLY=s3 \
+  -DBUILD_SHARED_LIBS=OFF \
+  -DENABLE_TESTING=OFF \
+  -DENABLE_UNITY_BUILD=ON
+cmake --build build/
+cmake --install build/
+popd
```
Comment (trxcllnt) [inline ci-wheel.Dockerfile:139]: Should `rm -rf /tmp/aws-sdk-cpp` here so it isn't included in this layer

**Pull request merged**